client/src/main/java/io/pravega/client/stream/impl/PositionInternal.java

```diff
@@ -16,15 +16,10 @@
 import java.util.Set;
 
 /**
- * A position has two components
- * 1. ownedSegments -- segments that can be read currently. Each ownedSegment also has an offset indicating the
+ * A position has ownedSegments -- segments that can be read currently. Each ownedSegment also has an offset indicating the
  * point until which events have been read from that segment. Completely read segments have offset of -1.
- * 2. futureOwnedSegments -- segments that can be read after one of the currently read segment is completely read. Each
- * segment in this set has exactly one previous segment that belongs to the set ownedSegments.
  * <p>
- * Well-formed position object. A position is called well-formed iff the following hold.
- * 1. for each segment s in futureOwnedSegment, s.previous belongs to ownedSegments and s.previous.offset != -1
- * 2. for each segment s in ownedSegment, s.previous does not belongs to ownedSegments
+ * Well-formed position object. A position is called well-formed iff each segment s in ownedSegment, s.previous does not belongs to ownedSegments
```

Comment:
This is trying to express this invariant, yes?

```
Obtaining a Position at any given point will return
a segment for all of the keyspace owned by the reader.
```

Reply:
yes. And the invariant that it will not contain multiple segments for the same portion of keyspace.